### ISSUE TITLE
Adept Active Skill reverting to base free ranks

### ISSUE THREAD

**Issue by imariaprime** [chummer5a/chummer5a#464]
Created an adept at Priority B, chose Exotic Melee Weapon as the free rank 4 active skill. Built the character, added two ranks to it bringing it up to rank 6. Enter Career mode, save, reopen.

Exotic Melee is now reset to rank 4.

Added the necessary karma as a refund to bring it back up to 6, assumed it was an error in conversion to Career mode. Saved, closed.

Opened it later. Exotic Melee is now rank 4 again.

I saved the file (opened with no changes) with a new name, and compared the two save files in a text editor. The only change was that the new file had reverted the rating 6 to 4. The karma log even still showed me buying up to rank 6, but opening the save reverts the skill to the base 4 free levels every time.

I edited the original save, changing the "freelevels" tag on Exotic Melee from 4 to 0. Now when it loads, the reversion doesn't occur.

The "Freelevels" tag is being incorrectly read, and is overwriting the total ranks on a skill.

**Comment by HaikenEdge** [comment]
This happens to grouped skills as well, actually. https://github.com/chummer5a/chummer5a/issues/428 discusses it briefly.

**Comment by imariaprime** [comment]
Peeking into clsCharacter.cs, it looks like broken skill groups are handled by setting the previous group ranks as "freelevels". So it's actually the same bug, but the problem is something about how "freelevels" are handled.

It's probably in SkillControl.cs somewhere, but I'm seriously amateur and this code is crazy-hard to follow.

**Comment by joha4270** [comment]
You are just as likely to be looking at clsUnique.cs.Skill or frmCareer.cs
(but i'm scared of those part of the codebase too, and i have been hacking
at it for half a year now :O)

But don't bother, i'm halfway through a total rewrite of skills, that
should fix all of those problems.

**Comment by imariaprime** [comment]
By this point, I'm just curious what is actually happening. There's a lot of times that I see freelevels is used as a minimum for a skill, and it gets manually set to the freelevels value if the total value is too low. That's probably being done wrong somewhere, hence the problem, but I'm not 100% sure how or where.

But if it's all being completely overwritten, then I suppose it's a moot point.

**Comment by chummer5a** [comment]
This should have been fixed in recent commits, I believe.
